api4/status_test.go

```diff
@@ -42,6 +43,30 @@ func TestGetUserStatus(t *testing.T) {
 		assert.Equal(t, "dnd", userStatus.Status)
 	})
 
+	t.Run("dnd status timed", func(t *testing.T) {
+		th.App.SetStatusDoNotDisturbTimed(th.BasicUser.Id, time.Now().Add(10*time.Minute).Format(time.RFC3339))
+		userStatus, resp := Client.GetUserStatus(th.BasicUser.Id, "")
+		CheckNoError(t, resp)
+		assert.Equal(t, "dnd", userStatus.Status)
+	})
+
+	t.Run("dnd status timed restore after time interval", func(t *testing.T) {
+		task := model.CreateRecurringTask("Unset DND Statuses", th.App.UpdateDNDStatusOfUsers, time.Duration(30)*time.Second)
+		th.App.SetStatusOnline(th.BasicUser.Id, true)
+		userStatus, resp := Client.GetUserStatus(th.BasicUser.Id, "")
+		CheckNoError(t, resp)
+		assert.Equal(t, "online", userStatus.Status)
+		th.App.SetStatusDoNotDisturbTimed(th.BasicUser.Id, time.Now().Add(1*time.Minute).Format(time.RFC3339))
+		userStatus, resp = Client.GetUserStatus(th.BasicUser.Id, "")
+		CheckNoError(t, resp)
+		assert.Equal(t, "dnd", userStatus.Status)
+		time.Sleep(90 * time.Second)
```

Comment:
This makes the test 90 seconds longer, what is a lot, we need something less time consuming here. I would try to do it in 1 second (2 seconds max) and check instantaneusly and then check again after the second passes.

Reply:
Done